catroid/src/main/java/org/catrobat/catroid/utils/ImportProjectsFromExternalStorage.java

```diff
@@ -0,0 +1,235 @@
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.utils;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.pm.ApplicationInfo;
+import android.os.AsyncTask;
+import android.preference.PreferenceManager;
+import android.support.v7.app.AlertDialog;
+import android.util.Log;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.FlavoredConstants;
+import org.catrobat.catroid.io.StorageOperations;
+import org.catrobat.catroid.ui.recyclerview.util.UniqueNameProvider;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.catrobat.catroid.common.FlavoredConstants.OLD_EXTERNAL_STORAGE_ROOT_DIRECTORY;
+import static org.catrobat.catroid.common.SharedPreferenceKeys.SHOW_COPY_PROJECTS_FROM_EXTERNAL_STORAGE_DIALOG;
+import static org.catrobat.catroid.io.StorageOperations.transferData;
+
+public class ImportProjectsFromExternalStorage implements DialogInterface.OnClickListener {
+	public static final String TAG = ImportProjectsFromExternalStorage.class.getSimpleName();
+	private static String pocketCodeEsFolderName = "Pocket Code";
+	private static String pocketCodeIsFolderName = "files";
+	private static String backpackFolderName = "backpack";
+	private Activity activity;
+
+	public ImportProjectsFromExternalStorage(Activity mActivity) {
+		activity = mActivity;
+	}
+
+	private static File copyProjectsFromExternalToInternalStorage(File srcDir, File dstDir) throws IOException {
+		UniqueNameProvider uniqueNameProvider = new UniqueNameProvider();
+		List<String> scope = new ArrayList<>();
+		if (!srcDir.exists()) {
+			throw new FileNotFoundException("Directory: " + srcDir.getAbsolutePath() + " does not exist.");
+		}
+
+		if (!srcDir.isDirectory()) {
+			throw new IOException(srcDir.getAbsolutePath() + " is not a directory.");
+		}
+		if (!dstDir.exists()) {
+			dstDir.mkdir();
+		}
+
+		if (!dstDir.isDirectory()) {
+			throw new IOException("Cannot create directory: " + dstDir.getAbsolutePath());
+		}
+
+		for (File srcFile : srcDir.listFiles()) {
+			String srcFileName = srcFile.getName();
+			if (srcFile.isDirectory()) {
+				boolean fileExists = false;
+				for (File dstFile : dstDir.listFiles()) {
```

Comment:
instead of using a boolean flag, put this for loop in a function, eg. directoryContainsFile(File directory, String filename)